src/AddOn.Episerver.Settings/Core/SettingsService.cs

```diff
@@ -355,8 +355,15 @@ public IEnumerable<ContentReference> SettingsRoots
             yield return settingsFromContent;
         }
 
-        var ancestors = ancestorReferencesLoader.GetAncestors(content.ContentLink);
+        var ancestors = ancestorReferencesLoader.GetAncestors(content.ContentLink)
+            .Where(x => !x.CompareToIgnoreWorkID(ContentReference.RootPage))
+            .ToList();
 
+        if (!ancestors.Contains(ContentReference.StartPage, ContentReferenceComparer.IgnoreVersion))
```

Comment:
Do we need to handle the case where start page is null, for instance in scheduled jobs? See https://world.optimizely.com/forum/developer-forum/CMS/Thread-Container/2017/5/how-can-i-get-a-reference-to-the-start-page-from-within-a-scheduled-task/

Reply:
Possibly, 
I think the code will run as is(but it's a bit stupid to add and try to load a null-reference)
But I don't think we should guess what is wanted (what start page to load when in a multi site, running as a scheduled job context), guessing the recommendation is to have a global setting as fallback if using it like this? or give a starting point in the page tree, in which the ancestor resolving will work out of the box.

Reply:
Hmm, good question. I think that a catalog might potentially be used on several sites. In the context of a scheduled job where you want to load settings for a commerce catalog item, I'm unsure how you want to handle this?

1. Just send in the current catalog item and just ignore the transition to CMS pages.
2. Send in the context of the page connecting the catalog.
3. Just ignore this since the settings that might be configured on the start page level is probably not interesting to get from a scheduled job context.

I suggest that we just keep the code as is - but that we add some documentation on how it works in a commerce context.